lib/fibonacci.rb

```diff
@@ -5,8 +5,18 @@
 # ....
 # e.g. 6th fibonacci number is 8
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: 0(n), where the program only loops through it n times (well, n - 2 times because the first few things are hardcoded cases), was wondering if it was better to do it with recursion? but that would run it with recursion instead of saving everything in an array and indexing it, but I think that would run more times?
+# Space complexity: constant, although you have to initialize space for an array
```

Comment:
⚠️Are you sure? That array looks like it's _linear_ to me.

I reckon you _can_ get to constant space though - any ideas on how you'd do it?